scripts/bootstrap.sh

```diff
@@ -158,54 +142,62 @@ BINARIES=true
 
 # Parse commandline args pull out
 # version and/or ca-version strings first
-if echo $1 | grep -q '\d'; then
-  VERSION=$1;shift
-  if echo $1 | grep -q '\d'; then
-    CA_VERSION=$1;shift
-    if echo $1 | grep -q '\d'; then
-      THIRDPARTY_IMAGE_VERSION=$1;shift
+if [ -n "$1" ] && [ "${1:0:1}" != "-" ]; then
+    VERSION=$1;shift
+    if [ -n "$1" ]  && [ "${1:0:1}" != "-" ]; then
+        CA_VERSION=$1;shift
+        if [ -n  "$1" ] && [ "${1:0:1}" != "-" ]; then
+            THIRDPARTY_IMAGE_VERSION=$1;shift
+        fi
     fi
-  fi
 fi
 
-# prior to 1.1.0 architecture was determined by uname -m
-if [[ $VERSION =~ ^1\.[0]\.* ]]; then
-  export FABRIC_TAG=${MARCH}-${VERSION}
-  export CA_TAG=${MARCH}-${CA_VERSION}
-  export THIRDPARTY_TAG=${MARCH}-${THIRDPARTY_IMAGE_VERSION}
+# prior to 1.2.0 architecture was determined by uname -m
+if [[ $VERSION =~ ^1\.[0-1]\.* ]]; then
+    export FABRIC_TAG=${MARCH}-${VERSION}
+    export CA_TAG=${MARCH}-${CA_VERSION}
+    export THIRDPARTY_TAG=${MARCH}-${THIRDPARTY_IMAGE_VERSION}
 else
-  # starting with 1.2.0, multi-arch images will be default
-  : ${CA_TAG:="$CA_VERSION"}
-  : ${FABRIC_TAG:="$VERSION"}
-  : ${THIRDPARTY_TAG:="$THIRDPARTY_IMAGE_VERSION"}
+    # starting with 1.2.0, multi-arch images will be default
+    : "${CA_TAG:="$CA_VERSION"}"
+    : "${FABRIC_TAG:="$VERSION"}"
+    : "${THIRDPARTY_TAG:="$THIRDPARTY_IMAGE_VERSION"}"
 fi
 
 BINARY_FILE=hyperledger-fabric-${ARCH}-${VERSION}.tar.gz
 CA_BINARY_FILE=hyperledger-fabric-ca-${ARCH}-${CA_VERSION}.tar.gz
 
 # then parse opts
-while getopts "h?db" opt; do
-  case "$opt" in
-    h|\?)
-      printHelp
-      exit 0
-    ;;
-    d)  DOCKER=false
-    ;;
-    b)  BINARIES=false
-    ;;
-  esac
+while getopts "h?dsb" opt; do
+    case "$opt" in
+        h|\?)
+            printHelp
+            exit 0
+            ;;
+        d)  DOCKER=false
+            ;;
+        s)  SAMPLES=false
+            ;;
+        b)  BINARIES=false
+            ;;
+    esac
 done
 
+if [ "$SAMPLES" == "true" ]; then
+    echo
+    echo "Clone hyperledger/fabric-samples repo"
+    echo
+    cloneSamplesRepo
+fi
 if [ "$BINARIES" == "true" ]; then
-  echo
-  echo "Installing Hyperledger Fabric binaries"
-  echo
-  binariesInstall
+    echo
+    echo "Pull Hyperledger Fabric binaries"
+    echo
+    pullBinaries
 fi
 if [ "$DOCKER" == "true" ]; then
-  echo
-  echo "Installing Hyperledger Fabric docker images"
-  echo
-  dockerInstall
+    echo
+    echo "Pull Hyperledger Fabric docker images"
+    echo
+    pullDockerImages
 fi
```

Comment:
Do we want to simplify this by just hardcoding the versions we want so we don't give the option for someone to have the wrong things installed?

Reply:
they're hardcoded at the top, when run with no args it does the right thing. I'd prefer to leave things as close to upstream as possible so we don't have to track changes.